mne/io/tests/test_reference.py

```diff
@@ -189,3 +190,37 @@ def test_set_bipolar_reference():
                   ch_info=[{'foo': 'bar'}, {'foo': 'bar'}])
     assert_raises(ValueError, set_bipolar_reference, raw,
                   'EEG 001', 'EEG 002', ch_name='EEG 003')
+
+
+@testing.requires_testing_data
+def test_add_reference():
+    raw = Raw(fif_fname, preload=True)
+    picks_eeg = pick_types(raw.info, meg=False, eeg=True)
+    # check if channel already exists
+    assert_raises(ValueError, add_eeg_reference, raw, raw.info['ch_names'][0])
+    # add reference channel
+    raw_ref = add_eeg_reference(raw, 'Ref', copy=True)
+    assert_equal(len(raw_ref._data), len(raw._data) + 1)
+    assert_array_equal(raw._data[picks_eeg, :], raw_ref._data[picks_eeg, :])
+
+    raw = add_eeg_reference(raw, 'Ref', copy=False)
+    assert_array_equal(raw._data, raw_ref._data)
+
+    ref_idx = raw.ch_names.index('Ref')
+    ref_data, _ = raw[ref_idx]
+    assert_array_equal(ref_data, 0)
+
+    # test add ref on epochs
+    raw = Raw(fif_fname, preload=True)
+    events = read_events(eve_fname)
+    picks_eeg = pick_types(raw.info, meg=False, eeg=True)
+    epochs = Epochs(raw, events=events, event_id=1, tmin=-0.2, tmax=0.5,
+                    picks=picks_eeg, preload=True)
+    epochs_ref = add_eeg_reference(epochs, 'Ref', copy=True)
+    assert_equal(len(epochs._data), len(epochs_ref._data))
+    ref_idx = epochs_ref.ch_names.index('Ref')
+    ref_data = epochs_ref.get_data()[:, ref_idx, :]
+    assert_array_equal(ref_data, 0)
+    picks_eeg = pick_types(epochs.info, meg=False, eeg=True)
+    assert_array_equal(epochs.get_data()[:, picks_eeg, :],
+                       epochs_ref.get_data()[:, picks_eeg, :])
```

Comment:
Lets test giving a list of names too, for example linked mastoids:

    add_eeg_reference(epochs, ['A1', 'A2'])